### DISCUSSION TITLE
How to get the stake address once my wallet is connected?

### DISCUSSION THREAD

**Discussion by chmod** [cardano-foundation/cardano-connect-with-wallet#94] [Q&A]
Once user connects their wallet, i'd like to get the stakeAddress and store it. 
I haven't been able to get it on the onConnect callback, as we can't use hooks within the callback I am sort of stuck. Any ideas?

**Comment by fabianbormann** [comment]
@chmod sorry for the late reply .. 

I don't know why but I do not get any notification for discussions in this repo. I hope it is not to late: 

You can probably use the `const { stakeAddress } = useCardano()` hook to get the stake address. You can use both at the same time. Hook & Button component. They will sync their state. 

Once the use connected the wallet using the button, the stakeAddress from `useCardano` will also change from null to "addr1....." you can use `useEffect(() =>  {...}, [ stakeAddress ])` for example to react on this change and use the stake address. If you have any other question feel free to join our Discord server. I usually look into new messages there: 

https://discord.com/channels/1022471509173882950/1114115748185514085 (and I will try to change my GH notification settings .. sorry again)
